test/unit/3p/test-3p-frame.js

```diff
@@ -602,23 +601,18 @@ describes.realWin('3p-frame', {amp: true}, (env) => {
         });
 
         it('should return null if the input is not a json', () => {
-          const errorStub = env.sandbox.stub(dev(), 'error');
           expect(deserializeMessage('amp-other')).to.be.null;
-          expect(errorStub).to.not.be.called;
         });
 
         it('should return null if failed to parse the input', () => {
-          const errorStub = env.sandbox.stub(dev(), 'error');
+          expectAsyncConsoleError(/MESSAGING: Failed to parse message/i, 2);
```

Comment:
/cc @rsimha to confirm this is correct.

Reply:
The usage on line 608 looks correct, but I don't see the next couple lines doing any waiting. Should there be an `await` to ensure that the asynchronous error doesn't leak from the test? Or are the operations synchronous, with the errors thrown asynchronously?

Reply:
It's sync, with the error thrown async.

Reply:
Thanks for clarifying. In that case, this code should work as written. Any issues will be caught by the "Local Unit Tests" job on CircleCI.

Reply:
It looks like the test is failing due to the error: https://app.circleci.com/pipelines/github/ampproject/amphtml/14320/workflows/8164eb80-8a86-4022-9fe8-6edb28601c2c/jobs/239541

This case expects that the function will return `null` even if given malformed json, which should cause the async error. Any tips on how to debug?

Reply:
Opened https://github.com/ampproject/amphtml/pull/35605 to fix.